File: app/src/main/java/me/imzack/app/ender/injector/component/AboutComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.AboutPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.AboutActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = AboutPresenterModule.class)
public interface AboutComponent {

    void inject(AboutActivity aboutActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/AboutComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.AboutPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.AboutActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(AboutPresenterModule::class))
interface AboutComponent {

    fun inject(aboutActivity: AboutActivity)
}
